### ISSUE TITLE
Enable batched sensitivity maps in NUFFT operator via n_batchs parameter

### ISSUE THREAD

**Issue by AsmaTANABEN** [mind-inria/mri-nufft#258]
When applying the NUFFT operator to multi-contrast or batched MRI data (assuming the same undersampling pattern across volumes), I set n_batchs = n_contrasts. In this context, it’s important that the NUFFT operator’s smaps setter supports input of shape (n_batchs, n_coils, *shape), since sensitivity maps vary across contrasts or batch elements.

Currently, I need to instantiate separate NUFFT operators for multi-contrast data and restrict the batch size to 1 for mono contrast dataset, which limits scalability. This feature would make the operator more versatile for multi-contrast reconstruction tasks and for training deep learning models with batch sizes greater than one.

**Comment by paquiteau** [comment]
I see your point, and its a valid one, however this would requires to rewrite the implementation for each backend (!), so not so easy to fix as is. Also note that the current batch implementation is sequential.

However smaps can be updated, so you have an easy workaround by defining something like that:

```python
class BatchedSmapsNufft(FourierOperatorBase):
    ... # init (self.nufft, self.batch_size, etc )
    def op(self.batched_smaps, batched_contrast):
        batched_kspace = self.xp.zero(self.batched_size, self.n_coils, self.nufft.n_samples)
        for b in range(self.batch_size):
            self.nufft.smaps = batch_smaps[b] # or a more lower level copy 
            batched_kspace[b] = self.nufft.op(batch_contrast[b])

       return batched_kspace
```

**Comment by AsmaTANABEN** [comment]
I agree that this isn't a simple fix. Regarding the suggested workaround, I'm a bit unsure how it would interact with backpropagation; specifically, how confident are we that this way of updating smaps  doesn’t interfere with gradient computation? Wouldn’t the operator use the last updated smaps during backprop, potentially leading to incorrect gradients for earlier batch elements? 
Please feel free to correct me if I’m mistaken. I'm bringing this point up because I'm not entirely sure how the current implementation handles this.

**Comment by paquiteau** [comment]
Backprop is likely going to be an issue indeed.  Then I think the `BatchedSmapsNufft` that I proposed above should instead be a `torch.nn.Module` with proper `forward` and `backward` methods. The smaps should be saved (with `ctx.save_for_backward` that is being called in the forward). 
Overall, you will have your own [autodiff.py](https://github.com/mind-inria/mri-nufft/blob/f24240c0e0a3a3d715fb9de1e3037a03ab6a8269/src/mrinufft/operators/autodiff.py) with an extra argument for the smaps
